pipeline/authz/keto_engine_acp_ory.go

```diff
@@ -115,9 +110,19 @@ func (a *AuthorizerKetoEngineACPORY) Authorize(r *http.Request, session *authn.A
 
 	var b bytes.Buffer
 	u := fmt.Sprintf("%s://%s%s", r.URL.Scheme, r.URL.Host, r.URL.Path)
+
+	action, err := rule.Replace(u, cf.RequiredAction)
```

Comment:
@aeneasr Do you intend to have this Replace functionality for regexp strategy only or for both regexp and glob?

Reply:
Hm, that is a good question. I'm not sure if we can actually use this for glob, right?

I also think that we should really update this handler and make it more generic. So it would be fine for me to just support this specific handler for regexp, and throw an error if glob is used (for now).

What do you think?

Reply:
I agree. I don't see the out-of-the-box replacement functionality for glob.

We can add a check for a flavour here: https://github.com/ory/oathkeeper/blob/master/pipeline/authz/keto_engine_acp_ory.go#L111 and return an error if it is not a blank string (defaults to regex), "regex" or "text". Otherwise, we invoke `rule.ReplaceAllString` and proceed as usual. We can rename this method to `RegexpReplaceAll` to show that currently only regexp replacement is supported. What do you think? 

What's the difference, by the way, between "text" and "regex"? It seems like both flavours invoke `regexp.ReplaceAllString`.

Reply:
> Hm, that is a good question. I'm not sure if we can actually use this for glob, right?
> 
> I also think that we should really update this handler and make it more generic. So it would be fine for me to just support this specific handler for regexp, and throw an error if glob is used (for now).
> 
> What do you think?

I agree. I don't see the out-of-the-box replacement functionality for glob.

We can add a check for a flavour here: https://github.com/ory/oathkeeper/blob/master/pipeline/authz/keto_engine_acp_ory.go#L111 and return an error if it is not a blank string (defaults to regex), "regex" or "text". Otherwise, we invoke `rule.ReplaceAllString` and proceed as usual. We can rename this method to `RegexpReplaceAll` to show that currently only regexp replacement is supported. What do you think? 

What's the difference, by the way, between "text" and "regex"? It seems like both flavours invoke `regexp.ReplaceAllString`.

Reply:
> return an error if it is not a blank string (defaults to regex), "regex" or "text". Otherwise, we invoke rule.ReplaceAllString and proceed as usual. We can rename this method to RegexpReplaceAll to show that currently only regexp replacement is supported. What do you think?
>
> What's the difference, by the way, between "text" and "regex"? It seems like both flavours invoke regexp.ReplaceAllString.

So the flavor is actually only relevant for the upstream service (ORY Keto), not for ORY Oathkeeper itself. Therefore, the flavor does not matter regarding the logic here - we can simply decide based on the access_rule matching strategy!

Reply:
ok, thank you, @aeneasr 

As we discussed, I added a new interface `MatchingEngine` and its glob implementation. 

I will some tests and update documentation later.